### PR TITLE
Problem: recent fixes not released

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- merge into the un-released rc2 veresion

### Updates
- fix several panic issues in ethermint
- remove authz module

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **New Features**
	- Support for hard-fork style upgrades introduced.
	- Integration test updated to include `push0` opcode functionality.

- **Updates**
	- Application upgraded to version v1.1.0-rc2.
	- Updated dependencies to the latest versions.

- **Bug Fixes**
	- Resolved an issue with concurrent write in fee history.

- **Refactor**
	- Removed the authz module for streamlining the application.

- **Chores**
	- Adjusted fee parameters in upgrade handlers for improved fee management.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,16 +1,14 @@
 # Changelog
 
-## UNRELEASED
+*January 1, 2023*
```
Comment (yihuang) [inline CHANGELOG.md:3]:
```suggestion
*January 2, 2023*
```

File: go.mod [hunk 2]
```diff
@@ -225,7 +225,7 @@ replace (
 	// Use cosmos keyring
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.2.0
 	github.com/cometbft/cometbft-db => github.com/crypto-org-chain/cometbft-db v0.0.0-20231011055109-57922ac52a63
-	github.com/cosmos/cosmos-sdk => github.com/crypto-org-chain/cosmos-sdk v0.46.0-beta2.0.20231221065437-7b1583073ee8
+	github.com/cosmos/cosmos-sdk => github.com/crypto-org-chain/cosmos-sdk v0.46.0-beta2.0.20240103063449-c887e863fae2
```
Comment (yihuang) [inline go.mod:228]: https://github.com/crypto-org-chain/cosmos-sdk/compare/7b1583073ee8...c887e863fae2

**Review by mmsqe — APPROVED** [review]

**Pull request merged**